mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js

```diff
@@ -0,0 +1,86 @@
+import { useWatch } from 'react-hook-form'
+import PropTypes from 'prop-types'
+import React, { useEffect, useState } from 'react'
+
+import { FormLabel, MenuItem, Stack, TextField } from '@mui/material'
+import { makeValidCamelCaseName } from '../../library/strings/makeValidCamelCaseName'
+import { makeValidClassName } from '../../library/strings/makeValidClassName'
+import { NestedQuestion } from '../../styles/forms'
+import { restorationAims as questions } from '../../data/questions'
+import language from '../../language'
+const stakeholdersPropType = PropTypes.arrayOf(
+  PropTypes.shape({ stakeholderType: PropTypes.string, stakeholderName: PropTypes.string })
+)
+
+const StakeholderBenefitsInputs = ({
+  optionId,
+  parentQuestionFieldName,
+  reactHookFormInstance,
+  stakeholders
+}) => {
+  const questionLabelId = `${optionId}-question-label`
+  const stakeholdersToUse = stakeholders ?? []
+  const {
+    setValue: setFormValue,
+    getValues: getFormValues,
+    control: formControl
+  } = reactHookFormInstance
+  const benefitsFieldName = `${parentQuestionFieldName}StakeholderBenefits`
+  const stakeholderBenefits = useWatch({ control: formControl, name: benefitsFieldName })
+
+  const handleOnChange = ({ event, stakeholderPropertyName }) => {
+    const previousBenefits = getFormValues(benefitsFieldName)
+
+    setFormValue(benefitsFieldName, {
+      ...previousBenefits,
+      [stakeholderPropertyName]: event.target.value
+    })
+  }
+  const menuItems = questions.nestedStakeholderBenefitsRanking.options.map((item) => (
+    <MenuItem key={item} value={item}>
+      {item}
+    </MenuItem>
+  ))
+
+  return (
+    <NestedQuestion>
+      <FormLabel id={questionLabelId}>
+        {questions.nestedStakeholderBenefitsRanking.question}
+      </FormLabel>
+      <div aria-labelledby={questionLabelId}>
+        {stakeholdersToUse.map(({ stakeholderType, stakeholderName }) => {
+          const selectInputId = makeValidClassName(`${optionId}${stakeholderType}-input`)
```

Comment:
this id ends up looking like aimNameStakeholderType

Reply:
and ends up being a property on the parent data. aimNameStakeholderType: values

Reply:
or probably nested within another property I willl call `stakeholderBenefits`. Not sure on if I implemented that or not yet.